File: backend/src/models/Suscripcion.py
from datetime import date

class Suscripcion:
    def __init__(self, id_suscripcion: int, costo: float, fecha_inicio: date, fecha_expiracion: date):
        self.id_suscripcion = id_suscripcion
        self.costo = costo
        self.fecha_inicio = fecha_inicio
        self.fecha_expiracion = fecha_expiracion
        self.activa = False

    def pagar(self) -> None:
        print(f"Pago de suscripción de ${self.costo} procesado.")

    def activar_suscripcion(self) -> None: 
        if self.fecha_expiracion >= date.today(): 
            self.activa = True
            print("Suscripción activada con éxito.")
        else:
            print("Error: La fecha de expiración ya pasó.")

    def __repr__(self) -> str: 
        return f"Suscripcion(ID='{self.id_suscripcion}', Activa='{self.activa}', Costo='${self.costo}')"

File: backend/src/models/Suscripcion.js
const mongoose = require('mongoose');

const SuscripcionSchema = new mongoose.Schema({
    usuario: { 
        type: mongoose.Schema.Types.ObjectId, 
        ref: 'Estudiante', 
        required: true 
    },
    costo: { 
        type: Number, 
        required: true 
    },
    fecha_inicio: { 
        type: Date, 
        default: Date.now 
    },
    fecha_expiracion: { 
        type: Date, 
        required: true 
    },
    activa: { 
        type: Boolean, 
        default: false 
    },
    pagado: {
        type: Boolean,
        default: false
    }
});


const Suscripcion = mongoose.model('Suscripcion', SuscripcionSchema);
module.exports = Suscripcion;